sphinx/util/__init__.py

```diff
@@ -245,18 +245,17 @@ def get_module_source(modname: str) -> Tuple[str, str]:
         raise PycodeError('error importing %r' % modname, err)
     filename = getattr(mod, '__file__', None)
     loader = getattr(mod, '__loader__', None)
+    try:  # prefer Native loader, as it respects #coding directive
```

Comment:
It is better to check loader is not None and loader has `get_source` method here.